aten/src/ATen/record_function.cpp

```diff
@@ -428,4 +428,55 @@ void RecordFunction::end() {
   active_ = false;
 }
 
+//
+namespace {
+std::atomic<int> global_record_all_functions_ {0};
+thread_local int tls_record_all_functions_ = 0;
```

Comment:
you can save on TLS accesses by grouping these two into a struct

Reply:
Yeah, this was also something that popped out to me when I saw this patch